core/ecschema-editing/src/Merging/EntityClassMerger.ts

```diff
@@ -2,40 +2,57 @@
+import { modifyClass } from "./ClassMerger";
+import { SchemaItemKey } from "@itwin/ecschema-metadata";
 
 /**
+ * Defines a merge handler to merge Entity Class schema items.
  * @internal
  */
-export default class EntityClassMerger extends ClassMerger<EntityClass> {
-  protected override async create(schemaKey: SchemaKey, ecClass: EntityClass): Promise<SchemaItemEditResults> {
-    return this.context.editor.entities.create(schemaKey, ecClass.name, ecClass.modifier);
-  }
+export const entityClassMerger: SchemaItemMergerHandler<EntityClassDifference> = {
+  async add(context, change) {
+    return context.editor.entities.createFromProps(context.targetSchemaKey, {
+      name: change.itemName,
+      schemaItemType: change.schemaType,
 
-  protected override async merge(itemKey: SchemaItemKey, changes: ClassChanges): Promise<SchemaItemEditResults> {
-    if (changes.entityMixinChanges.size > 0) {
-      if (changes.schemaItemMissing?.changeType !== ChangeType.Missing) {
-        return { errorMessage: `Changing the entity class '${itemKey.name}' mixins is not supported.`};
+      ...change.difference,
+    });
+  },
+  async modify(context, change, itemKey, item: MutableEntityClass) {
+    if(change.difference.mixins !== undefined) {
+      for(const mixin of change.difference.mixins) {
+        const mixinKey = await updateSchemaItemKey(context, mixin);
+        const result = await context.editor.entities.addMixin(itemKey, mixinKey);
+        if(result.errorMessage) {
+          return result;
+        }
       }
+    }
 
-      for (const change of changes.entityMixinChanges.values()) {
-        for (const entityMixinChange of change.entityMixinChange) {
-          const mixins = entityMixinChange.diagnostic.messageArgs! as unknown as [Mixin];
-          for (const mixin of mixins) {
-            const mixinKey = new SchemaItemKey(mixin.name, mixin.schema.schemaKey.matches(this.context.sourceSchema.schemaKey)
-              ? this.context.targetSchema.schemaKey
-              : mixin.schema.schemaKey);
-            const result = await this.context.editor.entities.addMixin(itemKey, mixinKey);
-            if (result.errorMessage !== undefined) {
-              return result;
-            }
-          }
-        }
+    return modifyClass(context, change, itemKey, item);
+  },
+};
+
+/**
+ * Defines a merge handler to merge Mixins to Entity Class schema items.
+ * @internal
+ */
+export const entityClassMixinMerger: SchemaItemMergerHandler<EntityClassMixinDifference> = {
+  async add(context, change) {
+    for(const mixinFullName of change.difference) {
+      const mixinKey = await updateSchemaItemKey(context, mixinFullName);
+      const entityKey = new SchemaItemKey(change.itemName, context.targetSchemaKey);
+      const result = await context.editor.entities.addMixin(entityKey, mixinKey);
+      if(result.errorMessage === undefined) {
+        throw new Error(result.errorMessage);
```

Comment:
shouldn't it throw if error message is defined?   Do we have a test that hits this code?